data/src/history/manager.rs

```diff
@@ -188,6 +190,98 @@ impl Manager {
         );
     }
 
+    pub fn record_chathistory_message(
+        &mut self,
+        server: &Server,
+        message: crate::Message,
+        subcommand: ChatHistorySubcommand,
+        message_reference: MessageReference,
+    ) {
+        self.data.add_chathistory_message(
+            server.clone(),
+            history::Kind::from(message.target.clone()),
+            message,
+            subcommand,
+            message_reference,
+        );
+    }
+
+    #[tokio::main]
+    pub async fn load_now(&mut self, server: Server, target: &str) {
+        let kind = if proto::is_channel(target) {
+            history::Kind::Channel(target.to_string())
+        } else {
+            history::Kind::Query(target.to_string().into())
+        };
+
+        let loaded_server = server.clone();
+        let loaded_kind = kind.clone();
+
+        let loaded_messages = history::load(&server.clone(), &kind.clone())
+            .map(move |result| Message::Loaded(loaded_server, loaded_kind, result))
+            .await;
+
+        self.update(loaded_messages);
+    }
+
+    #[tokio::main]
+    pub async fn make_partial_now(
+        &mut self,
+        server: Server,
+        target: &str,
+        message_reference: Option<MessageReference>,
+    ) {
+        let kind = if proto::is_channel(target) {
+            history::Kind::Channel(target.to_string())
+        } else {
+            history::Kind::Query(target.to_string().into())
+        };
+
+        self.data.map.get_mut(&server).and_then(|map| {
+            map.get_mut(&kind)
+                .and_then(|history| history.make_partial(message_reference))
+        });
+    }
```

Comment:
What are these `_now` methods used for? They're async but we're calling them from a blocking thread so we aren't actually doing anything.

Reply:
My understanding is that the [`#[tokio::main]`](https://docs.rs/tokio/latest/tokio/attr.main.html) allows them to run ~synchronously.  The histories are loaded, that much I am certain of.  Regardless, this is the crime against asynchronicity I was referring to earlier, and is a hack that's not really intended to stay.  The ultimate goal of this is to load the history such that the history can be used to (a) find a reference message to base the chathistory request on and (b) perform any necessary deduplication on received messages.  I've been working on reducing the need for (b) with the aim of ultimately removing that need entirely, but I haven't tested it lately to see if the need is gone yet.  I presume that (a) could be done more elegantly than I'm doing here, but I wanted to hash out some of the basic chathistory functionality before trying to work that out.